File: csharp/src/Scoreboard.cs
using System;
using System.Collections.Generic;

namespace IceCreamScorer
{
    public class Scoreboard
    {
        private readonly Scorer scorer;
        public Scoreboard(Scorer scorer)
        {
            this.scorer = scorer;
        }

        public void PrintScoreboard()
        {
            var names = new List<string>() { "Steve", "Julie", "Francis" };
            Console.WriteLine($"Scoreboard at time: {DateTime.Now:ddd MMM dd hh:mm:ss yyyy}");

            foreach (var name in names)
                PrintScore(name);
        }

        public void PrintScore(string name)
        {
            if (name == "Steve")
            {
                scorer.Flavour = IceCreamFlavours.Strawberry;
            }
            else
            {
                scorer.UpdateSelection();
            }
            int score = scorer.GetScore();
            Console.WriteLine($"{name} score: {score}");
        }
    }
}

File: python/src/scoreboard.py
import datetime

import scorer
from scorer import IceCream


def print_scoreboard():
    now = datetime.datetime.now()
    print_scoreboard_at(now)


def print_scoreboard_at(now):
    names = ["Steve", "Julie", "Francis"]
    print(f"Scoreboard at time {now}")
    for name in names:
        print_score(name)


def print_score(name):
    if name == "Steve":
        scorer.flavour = IceCream.Strawberry
    else:
        scorer.update_selection()
    score = scorer.get_score()
    print(f"{name} score: {score}")
